apps/vscode/package.json

```diff
@@ -1073,6 +1073,13 @@
           "default": true,
           "markdownDescription": "Reveal the preview panel after document render."
         },
+        "quarto.render.rPackageOutputDirectory": {
+          "order": 15,
+          "scope": "window",
+          "type": "boolean",
+          "default": true,
+          "markdownDescription": "Render output files in a temporary directory, when in an R package."
+        },
```

Comment:
This is new configuration, so users can turn off this behavior if they prefer:

![Screenshot 2024-05-08 at 4 08 13 PM](https://github.com/quarto-dev/quarto/assets/12505835/926f22dd-9223-4792-bdd6-85deef8be291)


I want to argue that this should be `true` by default, because otherwise, R package developers need to add all these output files to both `.gitignore` and `.Rbuildignore`.

Reply:
I would recommend doing a bit of work ahead of time to make sure that this is not going to create other bugs. In an R package, will users mostly preview `README.md` files and vignettes? In that case, I wonder if the right behavior is to use `--output-dir` only in these special cases. 

Like we discussed, relocating files for an .html target is unlikely to be robust because of things like CSS resource paths; in that case, we tend to recommend to use `--output-dir` together with [`embed-resources`](https://quarto.org/docs/output-formats/html-basics.html#self-contained)

Reply:
With that said, this is a pain in the neck to test in an automated fashion. It might be good enough to test this by making a mock test harness over at https://github.com/quarto-dev/quarto-cli. I can help with that; I think it's prudent to do it. Because `quarto-cli` won't generally know the ways in which the extension calls `quarto render`, `quarto preview` etc, we have a pretty big regression risk unless we do something about it.

Reply:
> In that case, I wonder if the right behavior is to use `--output-dir` only in these special cases.

I'm going to get some more feedback on this from folks, but my current belief is that there are basically _no cases_ where you want to check in generated code such as HTML, libs, etc, for an R package.

The one exception is `README.Rmd` being rendered to `README.md` but a) none of that works for people right now because of `output: github_document` and b) we can encourage them to use `devtools::build_readme()` for this purpose. If we want to be more forward thinking, we could _exclude_ `README.Rmd` and `README.qmd` from this treatment, rather than make a list of files to _include_ for this treatment.

Reply:
> I'm going to get some more feedback on this from folks, but my current belief is that there are basically no cases where you want to check in generated code such as HTML, libs, etc, for an R package.

I believe that! Really my only concern is that `quarto preview` itself might break under common use cases in R packages, and I'd hate for us to enable something by default that doesn't work well in these scenarios.